### ISSUE TITLE
Memory Cache Forever Issue

### ISSUE THREAD

**Issue by imconnorngl** [Julien-R44/bentocache#24]
When implementing a "forever" cache in using the `memoryDriver`. I recently came across some interesting behaviour, which led to me implementing the other caches in my program incorrectly causing caches to not be hit.

When using the `getOrSetForever` method. You are currently forced to provide `ttl: undefined` in the options, since `Infinity`. is not accepted by the `LRUCache` library. Without doing this the memory cache defaults to a super short ttl.
![image](https://github.com/Julien-R44/bentocache/assets/50084106/e81787e7-b25f-4504-84eb-cfdaaf3f6748)
![image](https://github.com/Julien-R44/bentocache/assets/50084106/876df1a8-0ff7-4c01-9118-32cff2066f36)

However, when using `ttl: undefined` on other adapters, such as the `knexDriver`. `ttl: undefined` is treated as no TTL, and that causes cache misses to occur, so when I implemented that driver I was experiencing cache misses, since it needs to have a default of `ttl: Infinity`
![image](https://github.com/Julien-R44/bentocache/assets/50084106/532dea11-2581-46e3-af98-0838c3264352)

**TLDR: The default TTL on the `memoryDriver` should be `undefined` rather than `Number.POSITIVE_INFINITY`.**

**Comment by Julien-R44** [comment]
Well I wasn't able to reproduce the error on my side. Using Infinity seems to work and doesn't throw any error
But, just in case, I still replaced `Infinity` with `undefined` for the memory driver since that's what's seems to be recommended by `node-lru-cache`.

Thanks for the issue!
